File: modules/file-transfer-core/src/main/java/com/shterneregen/securelan/filetransfer/service/FileTransferClientRequest.java
package com.shterneregen.securelan.filetransfer.service;

import java.nio.file.Path;
import java.util.Objects;

public record FileTransferClientRequest(
        String host,
        int port,
        String senderId,
        String recipientId,
        String sessionPassword,
        Path file
) {
    public FileTransferClientRequest {
        Objects.requireNonNull(host, "host must not be null");
        Objects.requireNonNull(senderId, "senderId must not be null");
        Objects.requireNonNull(recipientId, "recipientId must not be null");
        Objects.requireNonNull(sessionPassword, "sessionPassword must not be null");
        Objects.requireNonNull(file, "file must not be null");
        if (host.isBlank()) {
            throw new IllegalArgumentException("host must not be blank");
        }
        if (port < 1 || port > 65535) {
            throw new IllegalArgumentException("port must be between 1 and 65535");
        }
        if (senderId.isBlank()) {
            throw new IllegalArgumentException("senderId must not be blank");
        }
        if (recipientId.isBlank()) {
            throw new IllegalArgumentException("recipientId must not be blank");
        }
    }
}

File: modules/file-transfer-core/src/main/kotlin/com/shterneregen/securelan/filetransfer/service/FileTransferClientRequest.kt
package com.shterneregen.securelan.filetransfer.service

import java.nio.file.Path
import java.util.Objects

@JvmRecord
data class FileTransferClientRequest(
    val host: String,
    val port: Int,
    val senderId: String,
    val recipientId: String,
    val sessionPassword: String,
    val file: Path,
) {
    init {
        Objects.requireNonNull(host, "host must not be null")
        Objects.requireNonNull(senderId, "senderId must not be null")
        Objects.requireNonNull(recipientId, "recipientId must not be null")
        Objects.requireNonNull(sessionPassword, "sessionPassword must not be null")
        Objects.requireNonNull(file, "file must not be null")
        require(host.isNotBlank()) { "host must not be blank" }
        require(port in 1..65_535) { "port must be between 1 and 65535" }
        require(senderId.isNotBlank()) { "senderId must not be blank" }
        require(recipientId.isNotBlank()) { "recipientId must not be blank" }
    }
}
